packages/terra-button/src/Button.jsx

```diff
@@ -173,7 +173,7 @@ class Button extends React.Component {
     }
 
     // Add focus styles for keyboard navigation
-    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
+    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN || document.activeElement === event.currentTarget) {
```

Comment:
I think event.currentTarget will always match document.active element !! Curious are there any scenario where event.currentTarget would not be equal to document.activeElement?? 

If event.currentTarget always matches document.active element then probably you should update this condition to 
```js
 ( document.activeElement === event.currentTarget && ( event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN ) )
```

Reply:
Can some unit tests also be added to test the if statement and any potential edge cases?

Reply:
If we consider the case of terra-menu, upon Esc press, an opened menu closes and the programmatic focus is returned to button. This is triggering key event in button. If we update to suggested condition, it wouldn't pass as we didn't use return/tab key, thereby causing visible focus to fail in spite being the active element. Also mentioning that onfocus event isn't occurring in this case.